libs/ml/include/ml/model/model.hpp

```diff
@@ -99,6 +114,8 @@ class Model
                           DataType test_loss = fetch::math::numeric_max<DataType>());
 
 private:
+  friend class dmlf::distributed_learning::TrainingClient<TensorType>;
```

Comment:
Should this still be here? I thought shared_ptrs removed the need for the friend class.